### ISSUE TITLE
Design: TrajectoryAligner — alignment as a tier-2 LatentModule (not a Metric) + the cross-model mapper trigger

### ISSUE THREAD

**Issue by cmvcordova** [latent-reasoning-works/manylatents#280]
## Context

Infra for the **right-answer-wrong-reason tracing** work + the diffusion-geometry models. The detector compares a *clean* and a *misleading-hint* reasoning trajectory of the **same** problem; before any diffusion-geometry embedding (DiffusionMap/PHATE) can measure how they diverge, the two trajectories have to be put on a **common step axis** (they have different step counts). That alignment step is the piece designed here.

## Decision: alignment is a tier-2 `LatentModule`, not a `Metric`

The `Metric` protocol (`metrics/metric.py`) is single-stream, stateless, scalar-out — "run a metric on the embeddings." Alignment is categorically different: it consumes **two concurrent streams**, holds **fitted state** (the step-correspondence), and emits a **transformed trajectory** (the source re-expressed on the reference grid) that feeds the diffusion embed. That is the `fit`/`transform` tier:

- `fit(source, reference)` uses the base's existing `y` slot (`LatentModule.fit(x, y=None)`);
- `transform(source) -> aligned`; `fit_transform(source, reference)` is the single call the diffusion embed consumes;
- `residual(source, reference) -> per-step divergence curve` (the behavioral-baseline number falls out of this).

Memory stays bounded because the aligner operates on **pooled** step-states — the `ActivationSnapshot` (`lightning/activation_snapshot.py`) / `pool_hidden_states_per_step` reduction discipline — never raw per-token streams.

## The manyagents ↔ manylatents boundary

manylatents is the **public leaf**; it must not import manyagents. So the aligner never sees a `manyagents.ReasoningTrace` object. The many LLMs feed it through the existing seam:

```
manyagents: any adapter .run() -> ReasoningTrace (common schema across 11 adapters)
            local (hf/vllm) -> pool_hidden_states_per_step -> TraceStore (npz + jsonl)
── boundary: pooled tensors + metadata, NOT the Python object ──
manylatents: ReasoningTraceDataModule -> TrajectoryAligner.fit_transform(source, reference)
```

Only local adapters emit hidden states → only they feed the aligner; API adapters feed the behavioral table.

## Shipped (v0): #279

Same-model, `truncate`/`resample` matcher, `mapper=None`. `y=None` self-aligns (identity bound); cross-model dim mismatch is rejected. Known-answer battery passes (identity→0, truncate-prefix→0, resample-same-length identity, divergence localizes at the split step, non-negativity, torch type preserved).

## Deferred, each with an explicit trigger

- **Feature-space mapper (Procrustes/CCA).** Trigger: **cross-model** alignment (different hidden `d`) — the "many LLMs → aligner" requirement is the real second consumer that finally justifies the Mapping axis, with a clean `B = A@R ⇒ recover R` bound to test it. Not before.
- **DTW / OT matchers.** Trigger: trajectories whose reasoning length genuinely warps (v0 truncate/resample suffices while divergence is early).
- **Diffusion-native cost/residual** (diffusion-distance DTW cost, DSE / `diffusion_curvature` divergence). The research-novel v1 that ties the aligner to the diffusion-geometry models — deferred until v0's behavioral baseline exists to beat.

## What this supersedes / does not touch

- The **grand two-axis `Aligner`/`Alignment` taxonomy** was adversarially audited and **deferred as over-abstraction** at two cells — this issue is the disciplined outcome (one `LatentModule`, one matcher, seam open).
- `align_on_snapshot` (`algorithms/lightning/phase1_align.py`) is **left untouched** — the audit found zero production callers, so there was nothing to supersede. The one real DRY win it surfaced (the weighted-MSE-over-`layer_pairs` kernel duplicated at `phase1_align.py:178-183` and `distillation.py:279-289`) is a separate small cleanup, tracked apart from this.
